autofit/non_linear/search/optimize/pyswarms/search/abstract.py

```diff
@@ -15,6 +16,59 @@
 from autofit.plot.output import Output
 
 
+class FitnessPySwarms(Fitness):
+
+    def __call__(self, parameters, *kwargs):
+        """
+        Interfaces with any non-linear in order to fit a model to the data and return a log likelihood via
+        an `Analysis` class.
+
+        The interface is described in full in the `__init__` docstring.
+
+        `PySwarms` have a unique interface in that lists of parameters corresponding to multiple particles are
+        passed to the fitness function. A bespoke `__call__` method is therefore required to handle this.
+
+        The figure of merit is the log posterior multiplied by -2.0, which is the chi-squared value that is minimized
+        by the `PySwarms` non-linear search.
+
+        Parameters
+        ----------
+        parameters
+            The parameters (typically a list) chosen by a non-linear search, which are mapped to an instance of the
+            model via its priors and fitted to the data.
+        kwargs
+            Addition key-word arguments that may be necessary for specific non-linear searches.
+
+        Returns
+        -------
+        The figure of merit returned to the non-linear search, which is either the log likelihood or log posterior.
+        """
+
+        if isinstance(parameters[0], float):
+            parameters = [parameters]
+
+        figure_of_merit_list = []
+
+        for params_of_particle in parameters:
+
+            try:
+                instance = self.model.instance_from_vector(vector=params_of_particle)
+                log_likelihood = self.analysis.log_likelihood_function(instance=instance)
+                log_prior = self.model.log_prior_list_from_vector(vector=params_of_particle)
+                log_posterior = log_likelihood + sum(log_prior)
+                figure_of_merit = -2.0 * log_posterior
+            except exc.FitException:
+                figure_of_merit = -2.0 * self.resample_figure_of_merit

```

Comment:
You could set figure_of_metric = np.nan here to cause the if to evaluate to true and avoid repetition